android-build-tools/src/apk.rs

```diff
@@ -0,0 +1,191 @@
+            .map(Into::into)
+            .collect();
+        let permissions = metadata
+            .permission
+            .unwrap_or_default()
+            .into_iter()
+            .map(Into::into)
+            .collect();
+        let manifest = Manifest {
+            package_name: config.package_name,
+            package_label: config.package_label,
+            version_name: config.version_name,
+            version_code: config.version_code,
+            target_name: config.target_name,
+            target_sdk_version,
+            min_sdk_version: metadata.min_sdk_version.unwrap_or(23),
+            opengles_version: metadata.opengles_version.unwrap_or((3, 1)),
+            features,
+            permissions,
+            icon: metadata.icon,
+            fullscreen: metadata.fullscreen.unwrap_or(false),
+        };
+        Self {
+            ndk: config.ndk,
+            build_dir: config.build_dir,
+            assets: config.assets,
+            res: config.res,
+            manifest,
+        }
+    }
+
+    fn build_tool(&self, tool: &'static str) -> Result<Command, NdkError> {
+        let mut cmd = self.ndk.build_tool(tool)?;
+        cmd.current_dir(&self.build_dir);
+        Ok(cmd)
+    }
+
+    fn unaligned_apk(&self) -> PathBuf {
+        self.build_dir
+            .join(format!("{}-unaligned.apk", self.manifest.package_label))
+    }
+
+    fn apk(&self) -> PathBuf {
+        self.build_dir
+            .join(format!("{}.apk", self.manifest.package_label))
+    }
+
+    pub fn create_apk(&self) -> Result<UnalignedApk, NdkError> {
+        std::fs::create_dir_all(&self.build_dir)?;
+        self.manifest.write_to(&self.build_dir)?;
+
+        let mut aapt = self.build_tool("aapt")?;
+        aapt.arg("package")
+            .arg("-f")
+            .arg("-F")
+            .arg(self.unaligned_apk())
+            .arg("-M")
+            .arg("AndroidManifest.xml")
+            .arg("-I")
+            .arg(self.ndk.android_jar(self.manifest.target_sdk_version)?);
+
+        if let Some(res) = &self.res {
+            aapt.arg("-S").arg(res);
+        }
+
+        if let Some(assets) = &self.assets {
+            aapt.arg("-A").arg(assets);
+        }
+
+        aapt.status()?;
+
+        Ok(UnalignedApk(self))
+    }
+}
+
+pub struct UnalignedApk<'a>(&'a ApkConfig);
+
+impl<'a> UnalignedApk<'a> {
+    pub fn add_lib(&self, path: &Path, target: Target) -> Result<(), NdkError> {
+        if !path.exists() {
+            return Err(NdkError::PathNotFound(path.into()));
+        }
+        let abi = target.android_abi();
+        let file_name = path.file_name().unwrap();
+        let out = self.0.build_dir.join("lib").join(abi);
+        std::fs::create_dir_all(&out)?;
+        std::fs::copy(path, out.join(&file_name))?;
+
+        self.0
+            .build_tool("aapt")?
+            .arg("add")
+            .arg(self.0.unaligned_apk())
+            .arg(format!("lib/{}/{}", abi, file_name.to_str().unwrap()))
```

Comment:
Usually you should join the path components using `Path::join`.

Reply:
I'm not sure how aapt parses it on windows. It must be exactly this string for it to work, an absolute path for example breaks it